aiidalab_widgets_base/elns.py

```diff
@@ -0,0 +1,361 @@
+import ipywidgets as ipw
+import json
+import traitlets
+from pathlib import Path
+from aiida.orm import Node, QueryBuilder
+from IPython.display import clear_output, display
+from aiidalab_eln import get_eln_connector
+
+ELN_CONFIG = Path.home() / ".aiidalab" / "aiidalab-eln-config.json"
+
+
+def connect_to_eln(eln_name=None, **kwargs):
+    try:
+        with open(ELN_CONFIG, "r") as file:
```

Comment:
this fails, for example, if the `.aiidalab` folder does not exist.

Reply:
addressed in 041f5c822bc2612bf3b08a941e779825cb3442e1